docs/ai-core-APIs.md

```diff
@@ -0,0 +1,195 @@
+# Available APIs
+
+last updated: 24.07.2024
+
+✅ Available and tested
+
+❗Warning: The APIs not marked as tested in this section are in an experimental state. Use them at your own risk.
+
+# AI Core
+
+- ApplicationAPI
+- ArtifactApi ✅
+- ConfigurationApi ✅
+- DeploymentApi ✅
+- DockerRegistrySecretApi
+- ExecutableApi
+- ExecutionApi ✅
+- ExecutionScheduleApi
+- FileApi
+- KpiApi
+- MetaApi
+- MetricsApi
+- ObjectStoreSecretApi
+- RepositoryApi
+- ResourceApi
+- ResourceGroupApi
+- ResourceQuotaApi
+- ScenarioApi ✅
+- SecretApi
+- ServiceApi
+
+## ArtifactApi
+
+### Methods
+- [artifactQuery](../packages/ai-core/src/artifact-api.ts#L29)
+- [artifactCreate](../packages/ai-core/src/artifact-api.ts#L54)
+- [artifactGet](../packages/ai-core/src/artifact-api.ts#L73)
+- [artifactCount](../packages/ai-core/src/artifact-api.ts#L92)
```

Comment:
[minor] This feels a bit tedious to manually maintain. No strong opinion though. Maybe linking to the class itself is enough as a compromise?

Reply:
I wanted to add the specific lines because there is additional documentation in the code with description of parameters required for each method. I guess it depends on how often the client would change.

Reply:
I agree with Matthias about this being a bit too detailed for a README. A simple usage example (of the most common use case) should suffice.

Reply:
We should generate the API docs as well and then link them in the readme. This will save you the effort of describing every available function and it will be less error prone. Let's do that when we prepare an actual release.
I very much agree with Deeksha's comment, so that there is a general guidance on usage.